sfyra/pkg/capi/cluster.go

```diff
@@ -89,6 +89,14 @@ func NewCluster(ctx context.Context, metalClient runtimeclient.Reader, clusterNa
 		var endpoints []string
 
 		for _, machine := range machines.Items {
+			if !machine.DeletionTimestamp.IsZero() {
+				continue
+			}
+
+			if v1alpha3.MachinePhase(machine.Status.Phase) != v1alpha3.MachinePhaseRunning {
+				continue
+			}
+
```

Comment:
This the fix for the worker scaling flakes?

Reply:
yes, yes, and yes!

Reply:
Nice 😄

Reply:
I got fed up restarting the build, so I went ahead and fixed it hopefully.

Basically machine was marked as deleted, replicas in the status of machinedeployment is accurate, but we picked up stale info by looking into deleted machine